test/spec/Feature/Query/SingularSpec.hs

```diff
@@ -72,8 +72,7 @@ spec =
           `shouldRespondWith`
             [json|{"details":"The result contains 4 rows","message":"JSON object requested, multiple (or no) rows returned","code":"PGRST116","hint":null}|]
             { matchStatus  = 406
-            , matchHeaders = [ matchContentTypeSingular
-                             , "Preference-Applied" <:> "tx=commit" ]
+            , matchHeaders = [ matchContentTypeSingular ]
```

Comment:
These are the tests for part 1 of the fix, `tx=commit` no longer added.

Reply:
> Also this would be one very specific case of not sending that header... but would it be the same for all kinds of errors that are thrown by pg itself during the transaction?

Currently only the tx=commit would be included in pref-applied in case of an error. None of the others preferences get inside. To fix this we need to handle it in a different way than the current `optionalRollback`.

 > The preference applied header should be considered more of a "did the server understand the requested preference / value and would apply it if relevant"?
> When I set tx=commit and get a response without that preference applied... I would assume I am talking to either a PostgREST version that does not support this preference, yet (or anymore), o

That use case would be much better handled by `handling=strict` https://www.rfc-editor.org/rfc/rfc7240.html#section-4.4

Because currently pref-applied is unreliable, just on this PR it can be seen that `count=exact` was never applied. So I don't think any user (even us maintainers) should rely on just pref-applied to guarantee a PostgREST version understands certain preference. But `handling-strict` can be used to guarantee that.

Can work on `handling=strict` after this PR. I think the Plan module approach for `Prefer` I mentioned below would make the implementation simpler, we could build the Prefer and fail at the Plan level.

@wolfgangwalther How does that sound?

---

Merging this seems like a big win because of how much code is deleted (the prefer stuff was getting complicated).